e2e/tests/functional/plugins/plot/tagging.e2e.spec.js

```diff
@@ -0,0 +1,218 @@
+        //Alt+Shift Drag Start to select some points to tag
+        await page.keyboard.down('Alt');
+        await page.keyboard.down('Shift');
+
+        await canvas.dragTo(canvas, {
+            sourcePosition: {
+                x: 1,
+                y: 1
+            },
+            targetPosition: {
+                x: xEnd,
+                y: yEnd
+            }
+        });
+
+        //Alt Drag End
+        await page.keyboard.up('Alt');
+        await page.keyboard.up('Shift');
+
+        //Wait for canvas to stablize.
+        await canvas.hover({trial: true});
+
+        // add some tags
+        await page.getByText('Annotations').click();
+        await page.getByRole('button', { name: /Add Tag/ }).click();
+        await page.getByPlaceholder('Type to select tag').click();
+        await page.getByText('Driving').click();
+
+        await page.getByRole('button', { name: /Add Tag/ }).click();
+        await page.getByPlaceholder('Type to select tag').click();
+        await page.getByText('Science').click();
+    }
+
+    async function testTelemetryItem(page, canvas, telemetryItem) {
+        // Check that telemetry item also received the tag
+        await page.goto(telemetryItem.url);
+
+        await expect(page.getByText('No tags to display for this item')).toBeVisible();
+
+        //Wait for canvas to stablize.
+        await canvas.hover({trial: true});
+
+        // click on the tagged plot point
+        await canvas.click({
+            position: {
+                x: 325,
+                y: 377
+            }
+        });
+
+        await expect(page.getByText('Science')).toBeVisible();
+        await expect(page.getByText('Driving')).toBeHidden();
+    }
+
+    async function basicTagsTests(page, canvas) {
+        // Search for Science
+        await page.locator('[aria-label="OpenMCT Search"] input[type="search"]').click();
+        await page.locator('[aria-label="OpenMCT Search"] input[type="search"]').fill('sc');
+        await expect(page.locator('[aria-label="Search Result"]').nth(0)).toContainText("Science");
+        await expect(page.locator('[aria-label="Search Result"]').nth(0)).not.toContainText("Drilling");
+
+        // Delete Driving
+        await page.hover('[aria-label="Tag"]:has-text("Driving")');
+        await page.locator('[aria-label="Remove tag Driving"]').click();
+
+        await expect(page.locator('[aria-label="Tags Inspector"]')).toContainText("Science");
+        await expect(page.locator('[aria-label="Tags Inspector"]')).not.toContainText("Driving");
+
+        // Search for Driving
+        await page.locator('[aria-label="OpenMCT Search"] input[type="search"]').click();
+        await page.locator('[aria-label="OpenMCT Search"] input[type="search"]').fill('driv');
+        await expect(page.getByText('No results found')).toBeVisible();
+
+        //Reload Page
+        await Promise.all([
+            page.reload(),
+            page.waitForLoadState('networkidle')
+        ]);
+
```

Comment:
I'm noticing that sometimes the canvas click happens before the canvas is done rendering. We might be able to stabilize these a bit if we wait for the loading bar to disappear after reload:

```js
await page.locator('.l-view-section.c-progress-bar').waitFor({ state: 'detached' });
```

It's not as deterministic as we would like it to be but we can at least get a little bit closer to render time.